mcstatus/forge_data.py

```diff
@@ -0,0 +1,248 @@
+a response object. We support this implementation too.
+
+For more information see this file from forge itself:
+https://github.com/MinecraftForge/MinecraftForge/blob/54b08d2711a15418130694342a3fe9a5dfe005d2/src/main/java/net/minecraftforge/network/ServerStatusPing.java#L27-L73
+"""
+
+from __future__ import annotations
+
+import io
+from dataclasses import dataclass
+from typing import Final, IO, TYPE_CHECKING
+
+from mcstatus.protocol.connection import BaseConnection, BaseReadSync, Connection
+
+VERSION_FLAG_IGNORE_SERVER_ONLY: Final = 0b1
+IGNORE_SERVER_ONLY: Final = "<not required for client>"
+
+
+if TYPE_CHECKING:
+    from typing_extensions import NotRequired, Self, TypedDict
+
+    class RawForgeDataChannel(TypedDict):
+        res: str
+        """Channel name and ID (for example ``fml:handshake``)."""
+        version: str
+        """Channel version (for example ``1.2.3.4``)."""
+        required: bool
+        """Is this channel required for client to join?"""
+
+    class RawForgeDataMod(TypedDict):
+        modid: NotRequired[str]
+        modId: NotRequired[str]  # noqa: N815 # camel case
+        modmarker: str
+        """Mod version."""
+
+    class RawForgeData(TypedDict):
+        fmlNetworkVersion: int  # noqa: N815 # camel case
+        channels: list[RawForgeDataChannel]
+        mods: list[RawForgeDataMod]
+        d: NotRequired[str]
+        truncated: NotRequired[bool]
+
+else:
+    RawForgeDataChannel = dict
+    RawForgeDataMod = dict
+    RawForgeData = dict
+
+
+@dataclass
+class ForgeDataChannel:
+    res: str
+    """Channel name and ID (for example ``fml:handshake``)."""
+    version: str
+    """Channel version (for example ``1.2.3.4``)."""
+    required: bool
+    """Is this channel required for client to join?"""
+
+    @classmethod
+    def build(cls, raw: RawForgeDataChannel) -> Self:
+        """Build an object about Forge channel from raw response.
+
+        :param raw: ``channel`` element in raw forge response :class:`dict`.
+        :return: :class:`ForgeDataChannel` object.
+        """
+        return cls(res=raw["res"], version=raw["version"], required=raw["required"])
+
+    @classmethod
+    def decode(cls, buffer: Connection, mod_id: str | None = None) -> Self:
+        """Decode an object about Forge channel from decoded optimized buffer.
+
+        :param buffer: :class:`Connection` object from UTF-16 encoded binary data.
+        :param mod_id: Optional mod id prefix :class:`str`.
+        :return: :class:`ForgeDataChannel` object.
+        """
+        channel_identifier = buffer.read_utf()
+        if mod_id is not None:
+            channel_identifier = f"{mod_id}:{channel_identifier}"
+        version = buffer.read_utf()
+        client_required = buffer.read_bool()
+
+        return cls(
+            res=channel_identifier,
+            version=version,
+            required=client_required,
+        )
+
+
+@dataclass
+class ForgeDataMod:
+    modid: str
+    modmarker: str
```

Comment:
I don't think this is pythonic to name it so

```suggestion
    mod_id: str
    mod_marker: str
```

Reply:
Changed in 200c834